session/promise/promise_extra_data.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package promise
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/mysteriumnetwork/payments/promises"
+)
+
+// ExtraData represents the extra data in the promise
```

Comment:
We should add more info on what this extra data might be. For docs sake at least.

Reply:
This is exactly what it is - any extradata which will be hashed and validatable on service provider side, currently it can be service consumer identity, in case some 3d party signs promises on behalf of service consumer. Think of it more like a place holder for anything arbitrary. Though I agree that better docs needed

Reply:
Currently we send a consumer address, but we need more motivation for this, otherwise we just could have a separate field in a promise as ConsumerAddress

Reply:
It's not for sending info with promise. It's for validation. And no we don't send consumer address .- we put it in extra data and validate promise signature.

Reply:
True